mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoImplementationGenerator.java

```diff
@@ -50,6 +84,20 @@ protected String getFileName() {
   @Override
   protected JavaFile.Builder getContents() {
 
+    List<PartialClassGenerator> methodGenerators = new ArrayList<>();
+    for (Element child : interfaceElement.getEnclosedElements()) {
+      try {
+        if (child.getKind() == ElementKind.METHOD) {
+          ExecutableElement methodElement = (ExecutableElement) child;
+          if (methodElement.getAnnotation(EntityInjector.class) != null) {
+            methodGenerators.add(new EntityInjectorGenerator(methodElement, this, context));
+          }
+          // TODO handle other annotations
```

Comment:
Is there a need to handle other annotations here? I see that this class is called `DaoImplementationGenerator` so it should it handle only `@Dao` annotation, right? If we will want to handle other annotations here it will no be longer `DaoGenerator`.

Reply:
The generated class will implement all the methods declared on the interface, so we'll eventually handle all "query" annotations here.

```java
@Dao
public interface UserDao {
  @EntityInject
  void inject(User user, BoundStatement statement);

  @Insert
  void save(User user);

  @Select
  User findById(UUID id);
}
```
The loop iterates on the elements _inside_ `UserDao`.